mesg-foundation/engine
# Add new service exists api

fix https://github.com/mesg-foundation/engine/issues/1382

Add 2 new APIs for the services
- Exists(hash) that returns true if a service hash exists
- Hash(createServiceRequest) that returns the calculated hash of the service

Also, add support of queries with data. Every data sent should be encoded in proto.

**To Test**

You can call the api with grpcurl
```bash
grpcurl -proto protobuf/api/service.proto -plaintext -d "$(cat service.json)" localhost:50052 mesg.api.Service/Hash
```

This will return the hash for the service

```bash
grpcurl -proto protobuf/api/service.proto -plaintext -d "{\"hash\": \"xxxx\"}" localhost:50052 mesg.api.Service/Exists
```

This should return false and if you create the service and execute it again you should have true
